src/schema/v2/artwork/collectorSignals.ts

```diff
@@ -21,202 +19,158 @@ interface ActiveLotData {
   }
 }
 
+const AuctionCollectorSignals: GraphQLFieldConfig<any, ResolverContext> = {
+  resolve: async (artwork, {}, ctx) => {
+    const isInSale = artwork.sale_ids?.length > 0
+    if (
+      !checkFeatureFlag("emerald_signals-auction-improvements", ctx) ||
+      !isInSale
+    ) {
+      return null
+    }
+
+    const activeLotData = await getActiveSaleArtwork(
+      {
+        artworkId: artwork.id,
+        saleIds: artwork.sale_ids,
+      },
+      ctx
+    )
+
+    if (!activeLotData) {
+      return null
+    }
+
+    // Resolve all associated auctions data in one object
+    return {
+      artwork,
+      saleArtwork: activeLotData.saleArtwork,
+      sale: activeLotData.sale,
+    }
+  },
+
+  type: new GraphQLObjectType({
+    name: "AuctionCollectorSignals",
+    description: "Collector signals on a biddable auction lot",
+    fields: {
+      bidCount: {
+        type: new GraphQLNonNull(GraphQLInt),
+        description: "Bid count",
+        resolve: ({ saleArtwork }) => saleArtwork.bidder_positions_count ?? 0,
+      },
+      lotWatcherCount: {
+        type: new GraphQLNonNull(GraphQLInt),
+        description: "Lot watcher count",
+        resolve: ({ artwork }) => artwork.recent_saves_count ?? 0,
+      },
+      liveBiddingStarted: {
+        type: new GraphQLNonNull(GraphQLBoolean),
+        description: "Live bidding has started on this lot's auction",
+        resolve: ({ sale }) =>
+          !!sale.live_start_at && new Date(sale.live_start_at) <= new Date(),
```

Comment:
(still don't think checking timestamps with an MP server time is the way to go vs being able to rely on upstream/Gravity attributes - but can totally be iterated on)

Reply:
Yeah I hear you on this - maybe we could ticket it to look at some day in the future.